packages/edit-site/src/utils/get-closest-available-template.js

```diff
@@ -4,6 +4,20 @@
 import { find } from 'lodash';
 
 export default function getClosestAvailableTemplate( slug, templates ) {
+	if ( ! templates.length ) {
+		return {
+			id: 'index',
+			content: {
+				raw: '',
+			},
+			slug: 'index',
+			is_custom: false,
+			type: 'wp_template',
+			status: 'publish',
+			wp_id: null,
+		};
+	}
+
```

Comment:
This part allows the creation of a new template from the dashboard. If the theme doesn't have any templates, there's nothing to fallback to and the site-editor throws errors which prevents the user from creating a template.
This is a simple tweak to allow falling back to something - even if that something is an empty string.